lms/djangoapps/grades/tests/test_new.py

```diff
@@ -102,6 +104,19 @@ def setUp(self):
         }
         self.course.set_grading_policy(grading_policy)
 
+    def test_course_grade_no_access(self):
+        """
+        Test to ensure a grade can ba calculated for a student in a course, even if they themselves do not have access.
+        """
+        invisible_course = CourseFactory.create(visible_to_staff_only=True)
+        access = has_access(self.request.user, 'load', invisible_course)
+        self.assertEqual(access.has_access, False)
+        self.assertEqual(access.error_code, 'not_visible_to_user')
+
+        # with self.assertNoExceptionRaised: <- this isn't a real method, it's an implicit assumption
+        _ = CourseGradeFactory().create(self.request.user, invisible_course)
```

Comment:
Can you also verify that the resulting `CourseGrade` object has a grade percent of 0?